text/0026-rest-api/rest-api.yaml

```diff
@@ -0,0 +1,895 @@
+      responses:
+        200:
+          description: successful operation
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/GetInfoResponse'
+              examples:
+                Example 1:
+                  $ref: '#/components/examples/get-info-response-example-1'
+  /tips:
+    get:
+      tags:
+        - tangle
+      summary: Returns two non-lazy tips.
+      description: Returns two non-lazy tips. In case the node can only provide one tip, tip1 and tip2 are identical.
+      operationId: get-tips
+      responses:
+        200:
+          description: successful operation
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/GetTipsResponse'
+              examples:
+                Example 1:
+                  $ref: '#/components/examples/get-tips-response-example-1'
+                Example 2:
+                  $ref: '#/components/examples/get-tips-response-example-2'
+  /messages/by-hash:
+    get:
+      tags:
+        - messages
+      summary: Find a message by its hash.
+      description: Find a message by its hash. This endpoint returns the given
+        message including payload.
+      operationId: get-messages-by-hash
+      parameters:
+        - in: query
+          name: hashes
+          schema:
+            type: array
+            items:
+              type: string
+          required: true
+          description: Hash of the message.
+      responses:
+        "200":
+          description: successful operation
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/GetMessagesByHashResponse"
+              examples:
+                Example 1:
+                  $ref: '#/components/examples/post-messages-by-hash-response-example-1'
+        "400":
+          description: unsuccessful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+              examples:
+                default:
+                  $ref: '#/components/examples/invalid-data-provided-response-example'
+        "404":
+          description: unsuccessful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+              examples:
+                default:
+                  $ref: '#/components/examples/can-not-find-data-response-example'
+  /messages/by-tag:
+    get:
+      tags:
+        - messages
+      summary: Find all messages that are marked by a given tag.
+      description: Find all messages that are marked by a given tag.
+      operationId: get-messages-by-tag
+      parameters:
+        - in: query
+          name: hashes
```

Comment:
I believe this are tags not hashes. Specifically, there are no "tags" and the indexation payload uses the word "index" to substitute tags.